src/buildercore/terraform.py

```diff
@@ -56,6 +58,167 @@
 # see https://docs.fastly.com/guides/streaming-logs/changing-log-line-formats#available-message-formats
 FASTLY_LOG_LINE_PREFIX = 'blank' # no prefix
 
+# at the moment VCL snippets are unsupported, this can be worked
+# around by using a full VCL
+# https://github.com/terraform-providers/terraform-provider-fastly/issues/7 tracks when snippets could become available in Terraform
+FASTLY_MAIN_VCL_KEY = 'main'
+
+# TODO: extract classes and constant instances into buildercore.terraform.fastly module
+class FastlyVCL:
+    @classmethod
+    def from_string(cls, content):
+        return cls(content.splitlines())
+
+    def __init__(self, lines):
+        self._lines = lines
+
+    def __eq__(self, another):
+        return self._lines == another._lines
+
+    def __str__(self):
+        return "\n".join(self._lines)
+
+    def insert(self, section, statement):
+        # TODO: if we remove indentation altogether from all lines, this will become a single return value
+        section_start, indentation = self._find_section_start(section)
+        lines = list(self._lines)
+        lines.insert(section_start + 1, '')
+        lines.insert(
+            section_start + 1,
+            '%s  %s' % (indentation, statement)
+        )
+        return FastlyVCL(lines)
+
+    def _find_section_start(self, section):
+        lookup = r"(?P<indentation> +)sub vcl_%s {" % section
```

Comment:
not sure whether the snippets should be included before or after the #FASTLY macro. I embrace change